src/Shared/JSInterop/JSObjectReferenceJsonWorker.cs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Text.Json;
+
+namespace Microsoft.JSInterop
+{
+    /// <summary>
+    /// Used by JsonConverters to read or write a IJSObjectReference instance.
+    /// </summary>
+    internal static class JSObjectReferenceJsonWorker
```

Comment:
Should we consider making this `public` and in the `Microsoft.JSInterop.Implementation` namespace?

I'm thinking about other platform implementations like Mobile Blazor Bindings (and later, Blazor Desktop). They will also receive JSON blobs from JS interop that have the `__jsObjectId` on them, and also need to deserialize them to instances of their platform-specific `IJSObjectReference` type.

Reply:
Technically we could do that later, but then it may involve cross-repo collaboration on timings for that. If we just do it now we save ourselves a bunch of coordination later.